sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -325,7 +325,9 @@ case class Invoke(
 
   @transient lazy val method = targetObject.dataType match {
     case ObjectType(cls) =>
-      val m = cls.getMethods.find(_.getName == encodedFunctionName)
+      val m = cls.getMethods.find { m =>
```

Comment:
shall we use `filter` instead of `find`, and fail if more than one match is found?

Reply:
How about:

1. `getMethod` with param types
2. If not found, turn to find one method with parameter number
3. If more than one is found, or no one can be found, fails it

Reply:
SGTM

Reply:
+1 for that improved one.

Reply:
+1. Now I'm wondering whether `StaticInvoke` has the same issue since it uses `getDeclaredMethod`.